### ISSUE TITLE
Will the repeated dataset implementation be impacted by random choice with replacement?

### ISSUE THREAD

**Issue by CoinCheung** [open-mmlab/mmdetection#2043]
Hi, 

Thanks for the work of this open source project !!

I noticed that the implementation of the repeated dataset prolonged the return value of the method `__len__` of the subclass of `torch.utils.data.Dataset`, and meanwhile the index is computed as its remainder by the length of original dataset. 

My question is that the original dataset allows us to fetch batches that is without replacement, but the repeated dataset might not keep this. Since a shuffle method by `torch.randperm` will shuffle the index between each repeated version of the original dataset. Thus there might be one same training example sampled to one training batch. Will this be some problem worth taken care ??

For example, the original length of the dataset is 10, and it is repeated two times, thus the sampler can give us the indices 2 and 12 in the same batch which will produce a batch with two same training samples. Can we neglect this difference easily ?

**Comment by ZwwWayne** [comment]
The influence might be subtle in our experiments now.
If it affects the performance significantly, a customed dataset sampler could avoid this problem.

**Comment by CoinCheung** [comment]
I will try, thanks for support !!!
